app/src/main/java/com/gacode/relaunchx/ReLaunch.java

```diff
@@ -1746,6 +1676,34 @@ public boolean onTouch(View v, MotionEvent event) {
 		refreshBottomInfo();
 	}
 
+	private int getFirstItemToScrollUp(GridView gv) {
+		int viewHeight = gv.getHeight();
+		int targetItem = gv.getFirstVisiblePosition();
+			while (targetItem >= 0 && viewHeight > itemsHeights.get(targetItem)) {
```

Comment:
To make it right it's not that simple. There could be just one or two pixels of a line visible on the screen to account it as visible. In extreme example consider what if some item has let's say 20 lines but just one line is visible on the screen, still viewHeight would be of 20 lines.

Reply:
Yes, and my intention when scrolling up was that the first visible position (even 1 pixel height) to be the last visible position after scroll up.